mlrun/model_monitoring/controller.py

```diff
@@ -300,26 +313,38 @@ def _should_monitor_endpoint(
                 project=endpoint.metadata.project,
                 endpoint_id=endpoint.metadata.uid,
             ) as batch_window_generator:
-                base_period_seconds = base_period_minutes * _SECONDS_IN_MINUTE
-                if application_names != batch_window_generator.get_application_list():
+                current_time = mlrun.utils.datetime_now()
+                current_min_last_analyzed = (
+                    batch_window_generator.get_min_last_analyzed()
+                )
+                if (
+                    # Different application names, or last analyzed never updated
+                    application_names != batch_window_generator.get_application_list()
+                    or not current_min_last_analyzed
+                ):
```

Comment:
We should first check whether there are any applications before sending events to the worker.  
Regarding the `or not current_min_last_analyzed` condition, this may result in sending unnecessary events. When `current_min_last_analyzed` is `None`, it indicates there are no applications in the file. If applications do exist in the app list, they will already be handled by the first condition.

```suggestion
                if (
                    # Different application names, or last analyzed never updated
                    application_names != batch_window_generator.get_application_list()
                   and application_names
                ):
```